x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -2338,4 +2345,138 @@ public void testInInRangeValues() {
         PhysicalPlan p = testContext.optimizeAndPlan("SELECT long FROM test WHERE long IN (1, 2, 3, " + Long.MAX_VALUE + ", 5, 6, 7)");
         assertEquals(EsQueryExec.class, p.getClass());
     }
+
+    @SuppressWarnings("unchecked")
+    private static <T> List<T> metricAggsByField(Supplier<PhysicalPlan> physicalPlanSupplier, int fieldCount) {
+        PhysicalPlan p = physicalPlanSupplier.get();
+        assertEquals(EsQueryExec.class, p.getClass());
+        EsQueryExec ee = (EsQueryExec) p;
+        AggregationBuilder aggregationBuilder = ee.queryContainer().aggs().asAggBuilder();
+        assertEquals(fieldCount, ee.output().size());
+        assertEquals(ReferenceAttribute.class, ee.output().get(0).getClass());
+        assertEquals(fieldCount, ee.queryContainer().fields().size());
+        assertThat(fieldCount, greaterThanOrEqualTo(ee.queryContainer().aggs().asAggBuilder().getSubAggregations().size()));
+        Map<String, AggregationBuilder> aggsByName =
+            aggregationBuilder.getSubAggregations().stream().collect(Collectors.toMap(AggregationBuilder::getName, ab -> ab));
+        return IntStream.range(0, fieldCount).mapToObj(i -> {
+            String percentileAggName = ((MetricAggRef) ee.queryContainer().fields().get(i).v1()).name();
+            return (T) aggsByName.get(percentileAggName);
+        }).collect(Collectors.toList());
+    }
+
+    public void testPercentileNullOrUnspecifiedMethodOrDefaultMethodParameterIsTheSame() {
```

Comment:
not sure if this had been the target of a previous comment, but I would myself deduplicate the tests that only differ in the name of the function (like this and `testPercentileRankNullOrUnspecifiedMethodOrDefaultMethodParameterIsTheSame` or `testPercentileOptimization` and `testPercentileRankOptimization`) with a simple two steps loop.